var/spack/repos/builtin/packages/netcdf-cxx/package.py

```diff
@@ -19,9 +19,18 @@ class NetcdfCxx(AutotoolsPackage):
 
     depends_on('netcdf')
 
+    variant(
+        'netcdf4', default=True, description='Compile with netCDF4 support')
+
     @property
     def libs(self):
         shared = True
         return find_libraries(
             'libnetcdf_c++', root=self.prefix, shared=shared, recursive=True
         )
+
+    def setup_environment(self, spack_env, run_env):
+        if '+netcdf4' in self.spec:
+            # There is no clear way to set this via configure, so set the flag
+            # explicitly
+            spack_env.set('CPPFLAGS', '-DUSE_NETCDF4')
```

Comment:
This looks more like other stuff I've seen in Spack.  But my guess is you don't need `setup_environment` either.  See https://github.com/spack/spack/blob/1be258a587d0bed32de3df4312f6d8d0a9848f06/var/spack/repos/builtin/packages/wordnet/package.py

It will probably work with just:

```
    def configure_args(self):
        args = []
        if '+netcdf4' in self.spec:
            args.append('CPPFLAGS=-DUSE_NETCDF4')

        return args
```
Can you try that please and see if it works?

Reply:
It does work, is this preferable to `setup_environment()`? If so I'll add the change.

Reply:
Yes it is, thanks!

PS: Please understand that your efforts trying interim approaches were not in vain.  Your successful `setup_envrionment()` allowed me to search through existing Spack packages to find the more idiomatic way to do the same thing.

Reply:
Happy to help. Thank you for the suggestions!